lib/core/xskdev/xskdev.c

```diff
@@ -363,9 +371,20 @@ xskdev_rx_burst_default(void *_xi, void **bufs, uint16_t nb_pkts)
             xi->stats.rx_poll_wakeup++;
             (void)poll(&rxq->fds, 1, POLL_TIMEOUT);
         }
+
+        if (xi->idle_timestamp && cne_rdtsc_precise() > xi->idle_timestamp) {
+            if (poll(&rxq->fds, 1, xi->intr_timeout) == 0)
```

Comment:
Can you test this with a single thread handling more than one XSK? Make sure there is *no* traffic on all XSK except for one, and verify that the socket wakes up? I'm suspect this won't work when you have traffic on one socket, but not another, because you'll get stuck polling on the idle socket.

This was one of the reasons we changed the original POLL_TIMEOUT from 1000 to 0.

I think we should add all the FDs for all the AF_XDP sockets handled by a particular thread to an epoll() group, and then wait to get woken up. This won't work if the thread needs to handle non AF_XDP sockets (so if its using pktdev we need to make sure only pmd_af_xdp is used).

Reply:
A single thread polling more then one port will hang as you stated. The xskdev driver does not know about the other lports being polled by the same thread.

I believe this means the xskdev driver cannot be the point we do the poll() call and it needs to be at the thread level not the lport level. I am thinking this means we create library or functions to handle calling epoll() at the application level.

I know the developer could do this himself, but I think a set of APIs would make it easier to manage. This means the intr_timeout and idle_timeout values needs to be removed from jsonc lports and lport_groups an into the thread section.

We then create an epoll() adding each of the af_xdp socket fds and any other fd the thread wants to wait on. Need a bit more thought about this option. May need to figure out how to get the fd values for each lport.

Reply:
reworked how interrupts and file descriptors are handled.